web-server/v0.4/src/components/PageHeader/index.js

```diff
@@ -1,23 +1,49 @@
 import React, { PureComponent, createElement } from 'react';
 import PropTypes from 'prop-types';
-import pathToRegexp from 'path-to-regexp';
 import { Breadcrumb, Tabs, Tag } from 'antd';
 import classNames from 'classnames';
 import styles from './index.less';
 import urlToList from '@/components/_utils/pathTools';
 
 const { TabPane } = Tabs;
-export function getBreadcrumb(breadcrumbNameMap, url) {
-  let breadcrumb = breadcrumbNameMap[url];
-  if (!breadcrumb) {
+export const getBreadcrumb = (breadcrumbNameMap, url) => {
+  let breadcrumb = {};
+  if (breadcrumbNameMap[url]) {
+    breadcrumb = breadcrumbNameMap[url];
+  } else {
     Object.keys(breadcrumbNameMap).forEach(item => {
-      if (pathToRegexp(item).test(url)) {
-        breadcrumb = breadcrumbNameMap[item];
+      if (item.includes(url)) {
+        // Returns the route if final path correlates with current path
+        if (
+          item.split('/')[item.split('/').length - 1] === url.split('/')[url.split('/').length - 1]
```

Comment:
So this appears to execute the split 4 times instead of just twice.

Consider something like:

```
let item_comps = item.split('/');
let url_comps = url.split('/');
if (item_comps[item_comps.length - 1] === url_comps[url_comps.length - 1]) {
...
```

Reply:
Sounds great!